### PR TITLE
8295530: Update Zlib Data Compression Library to Version 1.2.13

### PR DESCRIPTION
**Pull request by LanceAndersen**
Hi all,

Please review this PR which will update the Zlib Data Compression Library from 1.2.11 to 1.2.13.

- I have run all of the Mach5 tiers which did not show any issues related to the upgrade.
- The JCK zip/jar tests also continue to pass.
- Jai ran his macOS aarch64 test(for [JDK-8282954](https://bugs.openjdk.org/browse/JDK-8282954)) with the zlib 1.2.13 port and has not seen the failure as seen with the MacOS aarch64 native zlib port

Best,
Lance

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295530](https://bugs.openjdk.org/browse/JDK-8295530): Update Zlib Data Compression Library to Version 1.2.13

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10773/head:pull/10773` \
`$ git checkout pull/10773`

Update a local copy of the PR: \
`$ git checkout pull/10773` \
`$ git pull https://git.openjdk.org/jdk pull/10773/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10773`

View PR using the GUI difftool: \
`$ git pr show -t 10773`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10773.diff">https://git.openjdk.org/jdk/pull/10773.diff</a>

</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
Can you confirm that this is exactly the 1.2.13 code, no patches/changes, right?

**Comment by LanceAndersen** [comment]
> Can you confirm that this is exactly the 1.2.13 code, no patches/changes, right?

That is correct,  there are no patches on top of the 1.2.13

**Review by AlanBateman — APPROVED** [review]
I didn't go through every file, instead I just sampled and compared to what is the zlib project. I think it's good to get this upgrade done.

File: src/java.base/share/native/libzip/zlib/README [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ZLIB DATA COMPRESSION LIBRARY
 
-zlib 1.2.11 is a general purpose data compression library.  All the code is
+zlib 1.2.13 is a general purpose data compression library.  All the code is
 thread safe.  The data format used by the zlib library is described by RFCs
 (Request for Comments) 1950 to 1952 in the files
-https://tools.ietf.org/html/rfc1950 (zlib format), rfc1951 (deflate format) and
+http://tools.ietf.org/html/rfc1950 (zlib format), rfc1951 (deflate format) and
```
Comment (jaikiran) [inline src/java.base/share/native/libzip/zlib/README:6]: Hello Lance, it appears that we had done this change to this file in the JDK as part of https://bugs.openjdk.org/browse/JDK-8283525. As Sean notes in that issue, the `http` link is back to working but `https` was still prefered during that change. Should we stick with `https`?

**Comment by jaikiran** [comment]
I checked out Mark Adler's upstream 1.2.13 tag https://github.com/madler/zlib/tree/v1.2.13 locally and then ran `diff src/java.base/share/native/libzip/zlib <root-of-1.2.13-of-madler-repo>`. The diff shows that for the (common) files that we have in this PR, there's no difference other than additional copyright headers.

Additionally against Mark's upstream git repo, I ran:

```
git diff --name-only --diff-filter=A v1.2.11 v1.2.13 
```
to find any newly added files (that might be relevant to us) between these 2 versions. The output is:
```
.github/workflows/cmake.yml
.github/workflows/configure.yml
.github/workflows/fuzz.yml
LICENSE
doc/crc-doc.1.0.pdf
examples/gznorm.c
examples/zran.h
```
None of which is relevant to JDK.

I also ran:
```
git diff --name-only --diff-filter=D v1.2.11 v1.2.13
```
to find any deleted files between these 2 versions and the output is:
```
contrib/amd64/amd64-match.S
contrib/asm686/README.686
contrib/asm686/match.S
contrib/inflate86/inffas86.c
contrib/inflate86/inffast.S
contrib/masmx64/bld_ml64.bat
contrib/masmx64/gvmat64.asm
contrib/masmx64/inffas8664.c
contrib/masmx64/inffasx64.asm
contrib/masmx64/readme.txt
contrib/masmx86/bld_ml32.bat
contrib/masmx86/inffas32.asm
contrib/masmx86/match686.asm
contrib/masmx86/readme.txt
```
none of which appear relevant to us.

Finally, I ran:
```
git diff --name-only --diff-filter=R v1.2.11 v1.2.13 
```
to find any renamed files. This returns no output (since nothing was renamed).

So overall, the change in this PR looks good to me and appears to be same as what's there in 1.2.13 of upstream.

**Review by jaikiran — APPROVED** [review]

File: src/java.base/share/native/libzip/zlib/README (hunk 1 above)
Comment (LanceAndersen) [reply src/java.base/share/native/libzip/zlib/README:6]: Hi Jai,

Given this is how it is in the README in the upstream repository and it is accessible, I would prefer not to change it and when/if we do, we should have the change made upstream

Comment (jaikiran) [reply src/java.base/share/native/libzip/zlib/README:6]: Thank you Lance, that sounds fine to me.

**Pull request closed without merging**